src/index.js

```diff
@@ -374,10 +402,63 @@ let accounts = [];
 let scrollToHandled = false;
 
 const isMetaMaskConnected = () => accounts && accounts.length > 0;
+let isWalletConnectConnected = false;
 
 // TODO: Need to align with @metamask/onboarding
 const isMetaMaskInstalled = () => provider && provider.isMetaMask;
 
+// test id
+const projectId = 'e6360eaee594162688065f1c70c863b7';
+
+const metadata = {
+  name: 'E2e Test Dapp',
+  description: 'This is the E2e Test Dapp',
+  url: 'https://metamask.github.io/test-dapp/',
+  icons: ['https://avatars.mywebsite.com/'],
+};
+
+const modal = createWeb3Modal({
+  ethersConfig: defaultConfig({ metadata }),
+  projectId,
+});
+
+async function handleWalletConnectChange({ isConnected }) {
+  if (isConnected) {
+    provider = modal.getWalletProvider().provider;
+    const providerDetail = {
+      info: {
+        uuid: provider.signer.uri,
+        name: 'wallet-connect',
+        icon: './wallet-connect.svg',
+        rdns: 'io.metamask',
+      },
+      provider,
+    };
```

Comment:
provider info that will be displayed on top of the test dapp

![Screenshot from 2023-12-12 12-39-35](https://github.com/MetaMask/test-dapp/assets/54408225/f2588707-efb1-466f-a64a-2131387f762a)